lib/autoproj/ops/configuration.rb

```diff
@@ -314,56 +314,56 @@ def inspect
                 to_s
             end
 
+            # Sort the package sets by dependency order
+            # Package sets that have no dependencies come first,
+            # the local package set (by main configuration) last
             def sort_package_sets_by_import_order(package_sets, root_pkg_set)
-                # The sorting is done in two steps:
-                #  - first, we build a topological order of the package sets
-                #  - then, we insert the auto-imported packages, following this
-                #    topological order, in the user-provided order. Each package is
-                #    considered in turn, and added at the earliest place that fits
-                #    the dependencies
-                topological = Array.new
-                queue = (package_sets.to_a + [root_pkg_set]).uniq
-                until queue.empty?
-                    last_size = queue.size
-                    pending = queue.dup
-                    queue = Array.new
-                    until pending.empty?
-                        pkg_set = pending.shift
-                        if pkg_set.imports.any? { |imported_set| !topological.include?(imported_set) }
-                            queue.push(pkg_set)
-                        else
-                            topological << pkg_set
-                        end
-                    end
-                    if queue.size == last_size
-                        raise ArgumentError, "cannot resolve the dependencies between package sets. There seem to be a cycle amongst #{queue.map(&:name).sort.join(', ')}"
-                    end
+                name_to_pkg_set = Hash.new
+                pkg_sets = Hash.new
+                all_dependencies = []
+
+                package_sets.each do |p|
+                    name_to_pkg_set[p.name] = p
```

Comment:
Is going through names really needed ? I would think that you should be using the package set object directly here (which would remove the need for `name_to_pkg_set`).